app/models/topical_event_about_page.rb

```diff
@@ -12,6 +12,9 @@ class TopicalEventAboutPage < ApplicationRecord
   validates_with SafeHtmlValidator
   validates_with NoFootnotesInGovspeakValidator, attribute: :body
 
+  after_save :republish_topical_event_to_publishing_api
+  after_destroy :republish_topical_event_to_publishing_api
```

Comment:
Could use `after_commit` here to cover both cases. Would also mean we're calling `after_update` though and I'm not 100% sure what that does...

https://apidock.com/rails/ActiveRecord/Transactions/ClassMethods/after_commit

Reply:
Thanks! 👍

I think `after_save` is the equivalent of `after_create` and `after_update`, so `after_commit` does seem appropriate here.

Also using `after_commit` will mean we're guaranteeing that the change is complete in the database before calling `Whitehall::PublishingApi.republish_async` which means we're making fewer assumptions about what that code does.

Reply:
Addressed in https://github.com/alphagov/whitehall/pull/7524/commits/de3e3bb23d473fd034d9197964202ac32e9717ea